### PR TITLE
Refactor Array utilities to extensions

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
GitHub notifications are delayed so CI won't kick in for a while...

File: okhttp/src/main/java/okhttp3/internal/Util.kt [hunk 1]
```diff
@@ -92,17 +92,16 @@ fun threadFactory(
...
+ * Returns an array containing only elements found in [this] and also in [other].
+ * The returned elements are in the same order as in [this].
  */
-fun intersect(
-  comparator: Comparator<in String>,
-  first: Array<String>,
-  second: Array<String>
+fun Array<String>.intersect(
+  other: Array<String>,
+  comparator: Comparator<in String>
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/Util.kt]: naturalOrder is a compelling default for these.

File: okhttp/src/main/java/okhttp3/internal/Util.kt [hunk 2]
```diff
@@ -113,21 +112,20 @@ fun intersect(
...
  * so small, and the likelihood of an intersection so great, that it is not worth the CPU cost of
  * sorting or the memory cost of hashing.
  */
-fun nonEmptyIntersection(
-  comparator: Comparator<String>,
-  first: Array<String>?,
-  second: Array<String>?
+fun Array<String>.hasIntersection(
+  other: Array<String>?,
+  comparator: Comparator<in String>
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/Util.kt]: naturalOrder is a compelling default for these.

Comment (JakeWharton) [reply okhttp/src/main/java/okhttp3/internal/Util.kt]: Yeah I was debating whether to do that or not. They're actually only used in one place where a custom comparator is always specified so I elected not to.

**Pull request merged**